src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1897,6 +1949,118 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
             }
         }
 
+        internal long AdoptOrphanedFor(DocumentsOperationContext context, string id)
+        {
+            using (DocumentIdWorker.GetSliceFromId(context, id, out var lowerId))
+            using (GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+            {
+                var collectionName = GetCollectionFor(context, lowerIdPrefix);
+                if (collectionName == null)
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                    return 0;
+                }
+
+
+                if (ShouldAdoptRevision(context, lowerId, lowerIdPrefix, collectionName, out var table, out var lastRevision))
+                {
+                    var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+                    var newEtag = _database.DocumentsStorage.GenerateNextEtag();
+                    var cv = _documentsStorage.GetNewChangeVector(context, newEtag);
+                    return CreateDeleteRevision(context, table, id, collectionName, cv, lastModifiedTicks, lastRevision.Flags) - 1;
+                }
+
+                return 0;
+            }
+        }
+
+        internal class TestingStuff
```

Comment:
let's move this under a partial class `RevisionsStorage.Debug`

Reply:
Cant do it,
You use `Debug.Assert` in this class so if we will create another class that called 'Debug' it will be too confusing.

Reply:
I mean let's move this to a file `RevisionsStorage.Debug.cs` that will have
```
public partial class RevisionsStorage
{
...
}
```
This shouldn't be a problem, I think

Reply:
Same problem, it 'collides' with another Debug class we use in RevisionStorage methods (Debug.Assert)

Reply:
Done